includes/menu.inc

```diff
@@ -5,25 +5,45 @@
  * Theme and preprocess functions for menus.
  */
 
+use Drupal\Core\Template\Attribute;
+
 /**
  * Implements hook_preprocess_menu().
  */
 function oe_bootstrap_theme_preprocess_menu(&$variables, $hook) {
   // No changes for menu toolbar.
-  if ($hook == 'menu__toolbar') {
+  if ($hook === 'menu__toolbar') {
     return;
   }
 
   // Get the current path.
   $current_path = \Drupal::request()->getRequestUri();
 
   $items = $variables['items'];
+
   foreach ($items as $key => $item) {
-    // If path is current_path, set active to li.
-    if ($item['url']->toString() == $current_path) {
-      // Add active link.
-      $variables['items'][$key]['attributes']['class'] = 'active';
+    $class = $item['in_active_trail'] ? ['active'] : [];
+    if (!empty($item['below'])) {
```

Comment:
this could be better, we could construct the menu item
$menu_item = [
  'label' => ..
  'path' => ..
  'attributes' => ..
  ..
]
$variables['items'][$key] = $menu_item
then if the item has below below
$variables['items'][$key] = [
  'trigger' => $menu_item,
] 
foreach ...

The idea is to set up the default values first, which is the more common use case, and only after handle the exception (the dropdown), this way the code becomes clearer and we dont have duplicate code like the one where we set label, path, attributes